.github/workflows/release.yml

```diff
@@ -22,6 +22,7 @@ jobs:
 
   draft-release:
     if: github.repository == 'anza-xyz/agave'
+    needs: [ trigger-buildkite-pipeline ]
```

Comment:
why do we need to wait for the `trigger-buildkite-pipeline` step 🤔 they won't need each other's data

Reply:
when the buildkite pipeline is triggered, the `trigger-buildkite-pipeline` step will show green immediately. it doesn't reflect the pipeline's result

Reply:
this might cause other issues. afaik it will wait for the whole pipeline to succeed. however, we ship bins in the buildkite pipeline, and if there’s no draft release, there’s nowhere to upload the bins

Reply:
OK, I've changed the workflow and the pipeline to run the publish test first and only then proceed with the rest.

Reply:
not saying my way is the only way. just putting an idea out there, and I’d love to hear yours too have you. have you considered running the crate check as a GitHub Action, or maybe we treating the audit check with the same level of blocking? I feel like using the build message to control conditional builds is a bit hacky, and it results in two builds for the same commit. that might make it less straightforward for people to pick the correct one or trace them.

Reply:
> have you considered running the crate check as a GitHub Action

I have, but github runners are under powered compared to our CI machines and this takes too long as it is. I'm also trying to keep the changes to the release pipeline in this PR to a minimum. 

> I feel like using the build message to control conditional builds is a bit hacky

It is a lot hacky, but changing the buildkite pipeline is only done manually through the UI right now, and I don't want to touch it. Not sure I understand your comment about 2 builds for the same commit though.

The goal for this PR is to get the crate check operational. I plan to follow up with another PR that addresses some pain points I see in the current setup.

Reply:
> I have, but github runners are under powered compared to our CI machines and this takes too long as it is. I'm also trying to keep the changes to the release pipeline in this PR to a minimum.

we can use more powerful GitHub-hosted machines to run it. do you have any link you can share? I’d like to know what time you got.

Reply:
> It is a lot hacky, but changing the buildkite pipeline is only done manually through the UI right now, and I don't want to touch it. Not sure I understand your comment about 2 builds for the same commit though.

oh, the 2 builds means that in this PR, the building process looks like
1. triggering the secondary pipeline with a special message
2. triggering the secondary pipeline again without the special message

so we will get 2 different build numbers in the buildkite page, but they are the same commits.

Reply:
> we can use more powerful GitHub-hosted machines to run it. do you have any link you can share? I’d like to know what time you got.

The last successful run from buildkite took 1h:

https://buildkite.com/anza/agave/builds/24657/steps/canvas?sid=019746d4-191b-45f9-ba67-f8af0a9415e3

We can use self hosted github runners I guess, but that's a big change.

Reply:
we can just use github larger runner atm I guess